server/src/main/java/org/candlepin/resource/PoolResource.java

```diff
@@ -267,6 +268,29 @@ public void deletePool(@PathParam("pool_id") String id) {
     }
 
     /**
+     * Retrieve a CDN for a Pool
+     *
+     * @return the cdn for a pool
+     * @httpcode 400
+     * @httpcode 200
+     */
+    @GET
+    @Path("{pool_id}/cdn")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Cdn getPoolCdn(
+        @PathParam("pool_id") @Verify(Pool.class) String id) {
+
+        Pool pool = poolManager.find(id);
+
+        if (pool == null) {
+            throw new NotFoundException(i18n.tr(
+                "Subscription Pool with ID ''{0}'' could not be found.", id));
+        }
+
+        return pool.getCdn();
```

Comment:
Do we want to throw an exception here if there is no CDN? That seems to be our general response to an entity not existing.